tests/optional_mutex/main.c

```diff
@@ -0,0 +1,125 @@
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "optional_mutex.h"
+#include "warnp.h"
+
+#define NUM_TIMES 100
+
+static pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
+static int value = 0;
+static int counter = 0;
+
+/* Wait duration can be interrupted by signals. */
+static inline void
+millisleep(size_t msec)
```

Comment:
I don't think it's worth importing `millisleep.h` from spiped to libcperciva just for this, but I'm happy to be corrected.